source/codegen/service_helpers.py

```diff
@@ -70,22 +89,19 @@ def create_standard_arg(parameter):
         type_without_brackets = common_helpers.get_underlying_type_name(
             parameter['type'])
         return f'({type_without_brackets}*){parameter_name}.data(), '
-    elif parameter['type'] in {"ViBoolean[]", "ViSession[]", "ViInt16[]"}:
+    elif _is_array_requiring_conversion(parameter):
```

Comment:
We don't know that it's an array yet though.

What about: `_is_array_that_requires_conversion`?

Could also be:

`elif is_array and _requires_conversion(parameter)`

Although my helper is probably not as comprehensive a list of all types that require conversion, so that could be misleading.